applications/ShapeOptimizationApplication/test_examples/01_Strain_Energy_Minimization_3D_Hook/run_optimization.py

```diff
@@ -58,43 +58,65 @@
 # ======================================================================================================================================
 
 class kratosCSMAnalyzer( (__import__("analyzer_base")).analyzerBaseClass ):
-    
-    # --------------------------------------------------------------------------    
+
+    # --------------------------------------------------------------------------
     def initializeBeforeOptimizationLoop( self ):
         self.__initializeGIDOutput()
         self.__initializeProcesses()
         self.__initializeSolutionLoop()
-        
-    # --------------------------------------------------------------------------
+
+# --------------------------------------------------------------------------
     def analyzeDesignAndReportToCommunicator( self, currentDesign, optimizationIteration, communicator ):
 
-         # Calculation of objective function
-        if communicator.isRequestingFunctionValueOf("strain_energy"):
-            
+         # Calculation of value of objective function
+        if communicator.isRequestingValueOf("strain_energy"):
+
             print("\n> Starting StructuralMechanicsApplication to solve structure")
             startTime = timer.time()
             self.__solveStructure( optimizationIteration )
             print("> Time needed for solving the structure = ",round(timer.time() - startTime,2),"s")
 
-            print("\n> Starting calculation of response value")
-            startTime = timer.time()                    
+            print("\n> Starting calculation of strain energy")
+            startTime = timer.time()
             listOfResponseFunctions["strain_energy"].CalculateValue()
-            print("> Time needed for calculation of response value = ",round(timer.time() - startTime,2),"s")
+            print("> Time needed for calculation of strain energy = ",round(timer.time() - startTime,2),"s")
 
-            communicator.reportFunctionValue("strain_energy", listOfResponseFunctions["strain_energy"].GetValue())  
+            communicator.reportValue("strain_energy", listOfResponseFunctions["strain_energy"].GetValue())
 
-        # Calculation of gradient of objective function
-        if communicator.isRequestingGradientOf("strain_energy"): 
+        # Calculation of value of constraint function
+        if communicator.isRequestingValueOf("mass"):
 
-            print("\n> Starting calculation of gradients")
-            startTime = timer.time()               
+            print("\n> Starting calculation of value of mass constraint")
```

Comment:
only use the type of the response ("mass") not how it is used in the optimization (constraint).